Automattic/jetpack
# AuthorRecommendation: add basic support

This PR builds upon the scaffolding of the blog.

The block now supports the attribute `recommendations`.
Based on this we render the recommendations for users via PHP.

The backend is not yet wired to this and will be done in other PR and diff. 

1721-gh-Automattic/dotcom-forge#issuecomment-1523158523

## Proposed changes:
<img width="1169" alt="Screenshot 2023-05-10 at 14 59 56" src="https://github.com/Automattic/jetpack/assets/7000684/da565de1-cf8f-46f6-b663-5bb9e7c66219">

* Add attribute recommendations.
* Block rendering in PHP.
* Editor: list subscriptions, and allow subscriptions to be selected. When selected the block attribute recommendations updates accordingly.

## Does this pull request change what data or activity we track or use?
The PR adds the recommendations (sites followed) as attributes of the block

### Other information:

## Jetpack product discussion
1721-gh-Automattic/dotcom-forge
pdDR7T-LK-p2

## Testing instructions:
1. Since the backend is not yet connected, this can be tested using mock data. You can apply what was removed here: https://github.com/Automattic/jetpack/commit/a15be4c760f3cc375c9adb8e6a2d2e905fa20086
2. Run `jetpack watch plugins/jetpack`
3. Add the block in the editor
4. Select a few of the recommendations and save
5. Visit the saved page to see if it is rendering properly
